catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/model/Search.java

```diff
@@ -172,6 +188,38 @@ private void updateStatus(String sourceId, QueryResponse queryResponse) {
                 State.SUCCEEDED :
                 State.FAILED));
         responseNum++;
+
+        queryResponse.getProcessingDetails()
+                .stream()
+                .filter(ProcessingDetails::hasException)
+                .forEach(details -> {
+                    status.addReason(generateSanitizedErrorMessage(details.getException(), 0));
+                });
+    }
+
+    private String generateSanitizedErrorMessage(Throwable e, int counter) {
+        if (e instanceof UnsupportedQueryException) {
```

Comment:
Could we put the list of exceptions to look for in a `Map` and do something like this?

```
private static final Map<Exception, String> knownExceptions = ImmutableMap.of(...);

// ...

Optional<String> reason knownExceptions.keySet().stream().parallel()
    .filter(knowException -> e instanceof knowException)
    .findFirst().map(knowException -> knowExceptions.get(knowException));

if (reason.isPresent()) {
    return reason.get();
}
```

Not sure if this the best way to do this using `stream`. @coyotesqrl may be able to improve this slightly :-)